operator/v1/types_network.go

```diff
@@ -788,3 +815,38 @@ const (
 	// between pods on the cluster network.
 	IPsecModeFull IPsecMode = "Full"
 )
+
+// +kubebuilder:validation:Enum:="";"Enabled";"Disabled"
+type RouteAdvertisementsEnablement string
+
+var (
+	// RouteAdvertisementsEnabled enables route advertisements for ovn-kubernetes
+	RouteAdvertisementsEnabled RouteAdvertisementsEnablement = "Enabled"
+	// RouteAdvertisementsDisabled disables route advertisements for ovn-kubernetes
+	RouteAdvertisementsDisabled RouteAdvertisementsEnablement = "Disabled"
+)
+
+// RoutingCapabilitiesProvider is a component providing routing capabilities.
+// +kubebuilder:validation:Enum=FRRK8s
+type RoutingCapabilitiesProvider string
+
+const (
+	// RoutingCapabilitiesProviderFRRK8s determines FRR-K8s is providing
+	// advanced routing capabilities.
+	RoutingCapabilitiesProviderFRRK8s RoutingCapabilitiesProvider = "FRRK8s"
+)
+
+// AdditionalRoutingCapabilities describes components and relevant configuration providing
+// advanced routing capabilities.
+type AdditionalRoutingCapabilities struct {
+	// providers is a set of enabled components that provide additional routing
+	// capabilities. Entries on this list must be unique. The  only valid value
+	// is currrently "FRRK8s" which provides FRR routing capabilities through
+	// the deployment of FRR-K8s.
+	// +listType=atomic
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))"
+	Providers []RoutingCapabilitiesProvider `json:"providers"`
```

Comment:
What would happen if I remove an item from this list?

Reply:
The corresponding component will be un-deployed from the cluster. In that process, the network should be reconfigured as it was before that component was deployed.